tests/acceptance/pageObjects/filesPage.js

```diff
@@ -56,6 +56,71 @@ module.exports = {
         .waitForOutstandingAjaxCalls()
         .useCss()
     },
+    /**
+     *
+     * @param {string} fileName
+     */
+    openSharingDialog: function (fileName) {
+      const shareBtnSelector = this.getFileRowSelectorByFileName(fileName) +
+        this.elements['shareButtonInFileRow'].selector
+
+      return this.initAjaxCounters()
+        .waitForFileVisible(fileName)
+        .useXpath()
+        .moveToElement(this.getFileRowSelectorByFileName(fileName), 0, 0)
+        .click(shareBtnSelector)
+        .waitForElementVisible('@sharingSideBar')
+        .useCss()
+    },
+    /**
+     *
+     * @param {string} input
+     */
+    enterAutoComplete: function (input) {
+      return this.initAjaxCounters()
+        .waitForElementVisible('@sharingAutoComplete')
+        .setValue('@sharingAutoComplete', input)
+    },
+    /**
+     *
+     * @param {string} pattern
+     */
+    sharingAutoCompleteShallContainElementsWith: function (pattern) {
```

Comment:
Please replace shall with should

Reply:
well - I did choose this method naming because it is not the filePage which should/shall do something - but the sharingAutoComplete.

I'm happy to get reasonable proposals - THX

Reply:
I suggested this because the behat step says `should be listed in the autocomplete list on the webUI`.
Also, shall denote in the future.

Reply:
In a way this is just a discussion about "legal" words. "should" has a bit more "wiggle-room", "shall" means it has to. "shall" is a more archaic word in ordinary modern English usage, it mostly happens in legal documents and sometimes in system specifications that are written in a "legal contract" way.
In Gherkin step text we have been using "should" - that makes the step more readable in ordinary English. The code underneath has asserts that impose a "shall" condition.
Take your pick, but be consistent.

Reply:
I would even go a step further and keep as much assertions as possible in the context.
have a function `getSharingAutoCompleteUsers()` or similar and do the assertion in the context, I think that is the correct place to do it. Reason: you might want to get the users from an other context and do some other assertion with it. the "Page" just does date input/output and the decision if the behaviour is correct or not is done externally (by the user or the context)